crates/cosmos/cosmos-chain-components/src/impls/queries/eip/feemarket.rs

```diff
@@ -0,0 +1,163 @@
+use cgp::core::error::CanRaiseError;
+use core::fmt;
+use core::ops::Div;
+use core::str::FromStr;
+use prost::DecodeError;
+use serde::Deserialize;
+use subtle_encoding::base64;
+
+use ibc_proto::cosmos::base::v1beta1::DecCoin;
+
+use crate::impls::queries::eip::dispatch::EipQueryError;
+use crate::traits::eip::eip_query::EipQuerier;
+use crate::traits::rpc_client::HasRpcClient;
+
+#[derive(Deserialize)]
```

Comment:
The constructs here can probably be deduplicated with `osmosis.rs`, and put in a module like `types.rs` in the same directory.

Reply:
Done in 1d284f40d7b7e18e0e29c9e155400a45f116978a